Frontend/ui-driver/src/Screens/DriverReferralScreen/View.purs

```diff
@@ -0,0 +1,335 @@
+                ][ textView
+                    [ width MATCH_PARENT
+                    , height WRAP_CONTENT
+                    , text $ getString YOUR_RANK
+                    , color "#FFFFFF"
+                    , textSize FontSize.a_12
+                    , gravity CENTER
+                    ]
+                , textView
+                    [ width MATCH_PARENT
+                    , height WRAP_CONTENT
+                    , text $ getString NOT_AVAILABLE_YET
+                    , color "#FFFFFF"
+                    , textSize FontSize.a_16
+                    , gravity CENTER
+                    , margin $ MarginTop 5
+                    ]
+                , textView
+                    [ width WRAP_CONTENT
+                    , height WRAP_CONTENT
+                    , text $ getString LEARN_MORE
+                    , background "#FFFFFF"
+                    , color "#2C2F3A"
+                    , cornerRadius 10.0
+                    , padding $ Padding 25 8 25 8
+                    , gravity CENTER
+                    , onClick push $ const $ LearnMore
+                    , margin $ MarginTop 5
+                    ]
+                 ]
+              ]
+     ]
+
+bottomNavBarView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+bottomNavBarView push state =
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  ][BottomNavBar.view (push <<< BottomNavBarAction) (navData ScreenNames.REFERRAL_SCREEN state.data.config.bottomNavConfig)]
+
+appQRCodeView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+appQRCodeView push state = 
+  linearLayout
+    [ width MATCH_PARENT
+    , height MATCH_PARENT
+    , gravity CENTER
+    , background Color.blackLessTrans
+    ][
+      linearLayout
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , orientation VERTICAL
+        , gravity CENTER
+        , cornerRadius 16.0
+        , background "#FFFFFF"
+        , margin $ MarginHorizontal 10 10
+        , padding $ Padding 12 12 12 12
+        ][ textView $
+            [ width WRAP_CONTENT
+            , height WRAP_CONTENT
+            , gravity CENTER
+            , text "Download Namma Yatri"
+            , margin $ MarginVertical 10 7
+            , color Color.black800
+            ] <> FontStyle.h2 TypoGraphy
+          , imageView
+            [ width $ V 280
+            , height $ V 280
+            , gravity CENTER
+            , imageWithFallback $ fetchImage FF_COMMON_ASSET "ny_driver_app_qr_code"
+            ]
+          , PrimaryButton.view (push <<< PrimaryButtonActionController state) (primaryButtonConfig state)
+         ]
+    ]
+
+qrScreenView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+qrScreenView push state = 
+  linearLayout
+    [ width MATCH_PARENT
+    , height WRAP_CONTENT
+    ][ imageView 
+            [ height $ V 0
```

Comment:
why is height and width 0?

Reply:
For not showing on the screen but we need it for shareImageMessageConfig

Reply:
😢